Consider floating point accuracy for system specs checks

h3lg3 opened the pull request:
Due to rounding issue checks for violating system specs in `make_trapezoid()` may fail, even though max values are not exceeded. For example the test below fails. Adding machine precision to the comparison solves the issue and fixes #267 

```
abs(amplitude2) > max_grad
125000.00000000001 > 125000
# ValueError: Refined amplitude (125000 Hz/m) is larger than max (125000 Hz/m).
```

The error can be raised using this example script

```
from pypulseq import Opts
from pypulseq import make_trapezoid

system = Opts(
    rf_dead_time=20e-6, 
    rf_ringdown_time=30e-6,
    grad_raster_time=1e-6,
    rf_raster_time=1e-6,
    block_duration_raster=1e-6,
    adc_raster_time=1e-6,
    max_grad=250e3 / 42.576e6 * 1e3,
    grad_unit='mT/m',
    max_slew=50,
    slew_unit='T/m/s',
    B0=50e-3,
    gamma=42.576e6,
)

grad = make_trapezoid(
    channel='x',
    area=140.625,
    system=system,
    max_grad=125000.0,
)

print(grad.amplitude)  # 125000.00000000001
```

FrankZijlstra commented:
Looks good. Could you also fix this for `make_arbitrary_grad`? I see the same issue remains there.

h3lg3 commented:
done! 
I also changed it in "make_extended_trapezoid_area" and changed the comparison form "grad >= max_grad" to "grad > max_grad" since a gradient as big as max_grad should be valid.

btasdelen commented:
@FrankZijlstra Do you think this is ready to be merged?

FrankZijlstra reviewed (approved):
All good. #272 should be considered for all the slew rate checks.
